packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -387,6 +398,14 @@ export class IndexedDbPersistence implements Persistence {
     return primaryStateListener(this.isPrimary);
   }
 
+  setDatabaseDeletedListener(
+    databaseDeletedListener: DatabaseDeletedListener
+  ): void {
+    this.databaseDeletedListener = async () => {
```

Comment:
Same feedback as below. Is there any value in retaining the `DatabaseDeletedListener` here? The actual events are emitted by IndexedDB so it seems like this is materially different from `primaryStateListener` above.